src/api/dirmonitor/fsevents.c

```diff
@@ -142,39 +123,28 @@ int translate_changes_dirmonitor(
 }
 
 
-int add_dirmonitor(struct dirmonitor_internal* monitor, const char* path) {
+int add_dirmonitor(struct dirmonitor_internal *monitor, const char *path) {
   stop_monitor_stream(monitor);
 
   monitor->lock = SDL_CreateMutex();
   pipe(monitor->fds);
 
   FSEventStreamContext context = {
-    .info = monitor,
-    .retain = NULL,
-    .release = NULL,
-    .copyDescription = NULL,
-    .version = 0
+    .info = monitor, .retain = NULL, .release = NULL, .copyDescription = NULL, .version = 0
   };
 
-  CFStringRef paths[] = {
-    CFStringCreateWithCString(NULL, path, kCFStringEncodingUTF8)
-  };
+  CFStringRef paths[] = { CFStringCreateWithCString(NULL, path, kCFStringEncodingUTF8) };
+
+  monitor->stream = FSEventStreamCreate(NULL,
+                                        stream_callback,
+                                        &context,
+                                        CFArrayCreate(NULL, (const void **) &paths, 1, NULL),
+                                        kFSEventStreamEventIdSinceNow,
+                                        0,
+                                        kFSEventStreamCreateFlagNone | kFSEventStreamCreateFlagWatchRoot
+                                          | kFSEventStreamCreateFlagFileEvents);
```

Comment:
A bit rough.
It would probably be better if the first argument was allowed to go to the new line, and the others get aligned to it.